echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/scm/GithubWebhookEventHandler.java

```diff
@@ -79,14 +76,22 @@ public void handle(Event event, Map postedEvent) {
       githubEvent = "push";
     }
 
-    String fullRepoName = webhookEvent.getFullRepoName(event, postedEvent);
+    String fullRepoName = webhookEvent.getFullRepoName();
     Map<String, String> results = new HashMap<>();
-    results.put("repoProject", webhookEvent.getRepoProject(event, postedEvent));
-    results.put("slug", webhookEvent.getSlug(event, postedEvent));
-    results.put("hash", webhookEvent.getHash(event, postedEvent));
-    results.put("branch", webhookEvent.getBranch(event, postedEvent));
-    results.put(
-        "action", githubEvent.concat(":").concat(webhookEvent.getAction(event, postedEvent)));
+    results.put("repoProject", webhookEvent.getRepoProject());
+    results.put("slug", webhookEvent.getSlug());
+    results.put("hash", webhookEvent.getHash());
+    results.put("branch", webhookEvent.getBranch());
+    results.put("action", githubEvent.concat(":").concat(webhookEvent.getAction()));
+    if (webhookEvent instanceof GithubPullRequestEvent) {
+      GithubPullRequestEvent pullRequestEvent = (GithubPullRequestEvent) webhookEvent;
+      if (((GithubPullRequestEvent) webhookEvent).getPullRequest() != null) {
```

Comment:
Use `pullRequestEvent` assigned above to remove the need to cast?

Reply:
omg, yes :p